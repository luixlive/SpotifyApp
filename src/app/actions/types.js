// Device reducer
export const DEVICE_TYPE_CHANGED = 'DEVICE_TYPE_CHANGED';

// User
export const LOAD_USER = 'LOAD_USER';
export const LOAD_USER_FAILED = 'LOAD_USER_FAILED';
export const LOAD_USER_SUCCEEDED = 'LOAD_USER_SUCCEEDED';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGOUT_USER_FAILED = 'LOGOUT_USER_FAILED';
export const LOGOUT_USER_SUCCEEDED = 'LOGOUT_USER_SUCCEEDED';

// UserStats
export const LOAD_USER_STATS = 'LOAD_USER_STATS';
export const LOAD_USER_STATS_FAILED = 'LOAD_USER_STATS_FAILED';
export const LOAD_USER_STATS_SUCCEEDED = 'LOAD_USER_STATS_SUCCEEDED';
