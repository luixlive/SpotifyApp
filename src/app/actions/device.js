import { DEVICE_TYPE_CHANGED } from './types';

export default deviceMobile => (
  { type: DEVICE_TYPE_CHANGED, payload: { deviceMobile } }
);
