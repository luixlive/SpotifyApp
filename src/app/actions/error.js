import { CLEAR_ERROR } from './types';

export default () => (
  { type: CLEAR_ERROR, payload: {} }
);
