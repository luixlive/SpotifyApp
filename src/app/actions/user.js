import { LOAD_USER, LOGOUT_USER } from './types';

export const loadUser = () => (
  { type: LOAD_USER, payload: { } }
);

export const logoutUser = () => (
  { type: LOGOUT_USER, payload: { } }
);
