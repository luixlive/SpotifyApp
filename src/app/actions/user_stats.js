import { LOAD_USER_STATS } from './types';

export default () => (
  { type: LOAD_USER_STATS, payload: { } }
);
