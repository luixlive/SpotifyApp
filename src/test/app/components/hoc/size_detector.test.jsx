import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import addRouterProps from './../../../test_utils/add_router_props';
import initialState from './../../../test_utils/initial_state';
import injectRouter from './../../../test_utils/inject_router';
import {
  sizeDetector as componentProps,
} from './../../../test_utils/components_props';
import SizeDetector, {
  getSizeDetector,
} from './../../../../app/components/hoc/size_detector';
import * as types from './../../../../app/actions/types';

describe('App Components HOC - SizeDetector', () => {
  describe('Snapshots', () => {
    it('renders', () => {
      const props = _.cloneDeep(componentProps);
      const SizeDetectorComponent = getSizeDetector(() => <div />);
      const rendered =
        renderer.create(<SizeDetectorComponent {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    it('should call deviceTypeChanged on resize events', () => {
      const deviceTypeChanged = jest.fn();
      const SizeDetectorComponent = getSizeDetector(() => <div />);
      mount(<SizeDetectorComponent deviceTypeChanged={deviceTypeChanged} />);

      global.innerWidth = 500;
      global.dispatchEvent(new Event('resize'));
      expect(deviceTypeChanged).toHaveBeenCalledTimes(1);

      global.innerWidth = 1000;
      global.dispatchEvent(new Event('resize'));
      expect(deviceTypeChanged).toHaveBeenCalledTimes(2);

      global.innerWidth = 700;
      global.dispatchEvent(new Event('resize'));
      expect(deviceTypeChanged).toHaveBeenCalledTimes(2);
    });

    it('shouldnt call deviceTypeChanged after componentWillUnmount', () => {
      const deviceTypeChanged = jest.fn();
      const SizeDetectorComponent = getSizeDetector(() => <div />);
      mount(<SizeDetectorComponent deviceTypeChanged={deviceTypeChanged} />)
        .unmount();

      global.innerWidth = 500;
      global.dispatchEvent(new Event('resize'));
      expect(deviceTypeChanged).toHaveBeenCalledTimes(0);

      global.innerWidth = 1000;
      global.dispatchEvent(new Event('resize'));
      expect(deviceTypeChanged).toHaveBeenCalledTimes(0);
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    const ComposedComponent = () => <div>Component</div>;
    let RenderedComponent;
    let store;
    let wrapper;
    beforeAll(() => {
      store = mockStore(initialState);
      RenderedComponent = SizeDetector(ComposedComponent);
      wrapper = mount(injectRouter(() => (
        <Provider store={store}>
          <RenderedComponent />
        </Provider>
      )));
    });

    it('renders', () => {
      expect(wrapper.find(RenderedComponent).length).toEqual(1);
    });

    it('matches initial state', () => {
      const props = addRouterProps(componentProps);
      expect(Object.keys(wrapper.find('SizeDetector').props()))
        .toEqual(Object.keys(props));
    });

    it('dispatches deviceTypeChanged', () => {
      expect(store.getActions().length).toBe(0);
      global.innerWidth = 500;
      global.dispatchEvent(new Event('resize'));

      const expectedAction = {
        type: types.DEVICE_TYPE_CHANGED,
        payload: { isDeviceMobile: true },
      };
      expect(store.getActions()[0]).toEqual(expectedAction);
    });
  });
});
