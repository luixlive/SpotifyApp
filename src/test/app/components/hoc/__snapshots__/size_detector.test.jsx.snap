// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components HOC - SizeDetector Snapshots renders 1`] = `<div />`;
